src/unity/python/turicreate/visualization/show.py

```diff
@@ -96,46 +96,302 @@ def show(x, y, xlabel="X", ylabel="Y", title=None):
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+
+    Return Value
+    ------------
+    An object of Plot that is the box and whiskers plot
+
+    Examples
+    --------
+    Make a box and whiskers plot.
+
+    >>> bp = turicreate.visualization.box_plot(tc.SArray(['a','b','c','a','a']),tc.SArray([4.0,3.25,2.1,2.0,1.0]))
+    """
+    if (type(x) != tc.data_structures.sarray.SArray or 
+        type(y) != tc.data_structures.sarray.SArray or
+        x.dtype != str or y.dtype not in [int, float]):
+        raise ValueError("turicreate.visualization.box_plot supports " + 
+            "x as SArray of dtype str and y as SArray of dtype: int, float." +
+            "\nExample: turicreate.visualization.box_plot(tc.SArray(['a','b','c','a','a']),tc.SArray([4.0,3.25,2.1,2.0,1.0]))")
     path_to_client = _get_client_app_path()
     title = _get_title(title)
     plt_ref = tc.extensions.plot_boxes_and_whiskers(path_to_client, x, y, 
       xlabel, ylabel, title)
     return Plot(plt_ref)
 
 def columnwise_summary(sf):
+    """
+    Plots a columnwise summary of the sframe provided as input, 
+    and returns the resulting Plot object.
+    
+    The function supports SFrames.
+
+    Parameters
+    ----------
+    sf : SFrame
+      The data to get a columnwise summary for
+    
+    Return Value
+    ------------
+    An object of Plot that is the columnwise summary
+
+    Examples
+    --------
+    Make a columnwise summary of an SFrame.
+
+    >>> x = turicreate.SArray([1,2,3,4,5])
+    >>> s = turicreate.SArray(['a','b','c','a','a'])
+    >>> sf_test = turicreate.SFrame([x,x,x,x,s,s,s,x,s,x,s,s,s,x,x])
+    >>> colsum = turicreate.visualization.columnwise_summary(sf_test)
+    """
+    if (type(sf) != tc.data_structures.sframe.SFrame):
+        raise ValueError("turicreate.visualization.columnwise_summary " + 
+            "supports SFrame")
     path_to_client = _get_client_app_path()
     plt_ref = tc.extensions.plot_columnwise_summary(path_to_client, sf)
     return Plot(plt_ref)
 
 def histogram(sa, xlabel="Values", ylabel="Count", title=None):
+    """
+    Plots a histogram of the sarray provided as input, and returns the 
+    resulting Plot object.
+    
+    The function supports numeric SArrays with dtypes int or float.
+
+    Parameters
+    ----------
+    sa : SArray
+      The data to get a histogram for. Must be numeric (int/float)
+    xlabel : str (optional)
+      The text label for the X axis. Defaults to "Values".
+    ylabel : str (optional)
+      The text label for the Y axis. Defaults to "Count".
+    title : str (optional)
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+    
+    Return Value
```

Comment:
Same comment as for scatter plot (should be Returns, with name/type/description in special syntax).